test cases/common/145 special characters/meson.build

```diff
@@ -35,3 +35,36 @@ gen2 = custom_target('gen2',
   output : 'result2',
   install : true,
   install_dir : get_option('datadir'))
+
+# test that we can pass these special characters in compiler arguments
+#
+# (this part of the test is crafted so we don't try to use these special
+# characters in filenames or target names)
+#
+# TODO: similar tests needed for languages other than C
+
+special = [['dollar', '$'],
+           ['hash', '#'],
+           ['quote', '\''],
+           ['doublequote', '"'],
+           ['lt', '<'],
+           ['gt', '>'],
+           ['amp', '&'],
+           ['slash', '/'],
+           ['backslash', '\\']]
+
+foreach s : special
+  args = '-DCHAR="@0@"'.format(s[1])
```

Comment:
Maybe use a conditional around each of these, and avoid trying to quote " or ' with itself?